File: backend/FoodMatch/src/FoodMatch.Domain/Entities/Food.cs
using FoodMatch.Domain.Enums;

namespace FoodMatch.Domain.Entities;

/// <summary>
/// Represents a food item in the system.
/// Maps to the 'foods' table.
/// </summary>
public class Food
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public FoodCategory? Category { get; set; }
    public string? CuisineType { get; set; }
    public string? ImageUrl { get; set; }
    public PriceRange? AvgPriceRange { get; set; }
    public MealTime? MealTime { get; set; }
    public bool IsActive { get; set; } = true;

    // Navigation properties
    public ICollection<FoodFoodTag> FoodFoodTags { get; set; } = new List<FoodFoodTag>();
    public ICollection<FoodSuggestionLog> SuggestionLogs { get; set; } = new List<FoodSuggestionLog>();
}

File: frontend/src/types/food.ts
export interface FoodResponseDto {
  id: number;
  name: string;
  category: string;
  cuisineType: string;
  avgPriceRange: string;
  mealTime: string;
  tags: string[];
}

export interface FoodRandomRequestDto {
  category?: string;
  sessionId: string;
}

export interface FoodSuggestRequestDto {
  sessionId: string;
  mealTime?: string;
  priceRange?: string;
  includedTags?: string[];
  excludedTags?: string[];
}
